zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/util/XmlUtil.java

```diff
@@ -51,16 +71,66 @@ public String getFlowNodeName(final FlowNodeInstanceEntity flowNode) {
         flowNode.flowNodeId());
   }
 
+  public HashMap<Long, Map<Long, String>> getUserTasksNames(final List<UserTaskEntity> userTasks) {
+    return getNamesForEntities(
+        userTasks,
+        UserTaskEntity::processDefinitionKey,
+        UserTaskEntity::userTaskKey,
```

Comment:
I think it is better that you pass the task elementId as all tasks sharing the same elementId+processDefinitionKey have the same name.
Then you return `Map<Long, Map<String, String>>` where the key of the second map is elementId
Same thing for flowNodeInstances
This way you parse the xml once per elementId of flowNodeId